planning/autoware_path_optimizer/src/node.cpp

```diff
@@ -361,18 +378,40 @@ std::vector<TrajectoryPoint> PathOptimizer::optimizeTrajectory(const PlannerData
   }();
   replan_checker_ptr_->updateData(planner_data, is_replan_required, now());
   if (!is_replan_required) {
+    mrm_required_ = false;
     return getPrevOptimizedTrajectory(p.traj_points);
   }
 
   if (enable_skip_optimization_) {
+    mrm_required_ = false;
```

Comment:
Instead of inserting `mrm_required_ = false;` one by one, let's put it at the beginning of the `optimizeTrajectory` function only once.